client/components/app-promo/lib/promo-retriever.js

```diff
@@ -0,0 +1,34 @@
+import { sample } from 'lodash';
```

Comment:
The containing `lib` folder feels out of place as far as folder structure goes. It should probably be moved into the `client/lib` directory. If it's very specifically tied to the component currently using it (which we should be cautious not to always assume), it could be inlined top-of-file or perhaps moved to a file adjacent in the component directory (though this is probably still generally discouraged).